src/test/kotlin/com/exactpro/th2/lwdataprovider/TestSseEventSearchRequest.kt

```diff
@@ -1,93 +1,162 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
 package com.exactpro.th2.lwdataprovider
 
 import com.exactpro.cradle.TimeRelation
+import com.exactpro.th2.common.grpc.EventID
 import com.exactpro.th2.dataprovider.grpc.EventSearchRequest
 import com.exactpro.th2.lwdataprovider.entities.exceptions.InvalidRequestException
 import com.exactpro.th2.lwdataprovider.entities.requests.SseEventSearchRequest
+import com.exactpro.th2.lwdataprovider.entities.requests.SseMessageSearchRequest
 import com.google.protobuf.Timestamp
 import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.assertThrows
 
 class TestSseEventSearchRequest {
-    @Test
-    fun testConstructorParam(){
-        // startTimestamp - 1, endTimestamp - 2, searchDirection - AFTER (default)
-        var params = mutableMapOf("startTimestamp" to listOf("1"),
-            "endTimestamp" to listOf("2", "3"))
-        var eventSearchReq = SseEventSearchRequest(params)
-        Assertions.assertNotNull(eventSearchReq.startTimestamp)
-        Assertions.assertNotNull(eventSearchReq.endTimestamp)
-        Assertions.assertEquals(TimeRelation.AFTER, eventSearchReq.searchDirection)
 
-        // startTimestamp and resumeFromId - nulls
-        params.remove("startTimestamp")
-        params.remove("endTimestamp")
-        assertThrows<InvalidRequestException> {
-            eventSearchReq = SseEventSearchRequest(params)
+    @Nested
+    inner class TestConstructorParam(){
+        @Test
+        // when startTimestamp and resumeFromId - nulls, should throw exception
+        fun testEmptyParamsMap(){
+            assertThrows<InvalidRequestException> {
+                SseEventSearchRequest(mapOf())
+            }
         }
 
-        // resumeFromId != null - testing checkEndTimestamp validation from here till the end
-        params["resumeFromId"] = listOf("1")
-        eventSearchReq = SseEventSearchRequest(params)
-        Assertions.assertNull(eventSearchReq.startTimestamp)
-        Assertions.assertNull(eventSearchReq.endTimestamp)
+        @Test
+        // resumeFromId != null, startTimestamp and endTimestamp - nulls
+        fun testTimestampNullsResumeIdNotNull(){
+            val eventSearchReq = SseEventSearchRequest(mapOf("resumeFromId" to listOf("1")))
+            Assertions.assertNull(eventSearchReq.startTimestamp)
+            Assertions.assertNull(eventSearchReq.endTimestamp)
```

Comment:
When you write tests it should be clear what is going on. In most `assert*` method there is a parameter with a message or message supplier. It will be called if the assertion is failed. It should provide information about what has failed.

Here for example you can add this inforamtion
```suggestion
            Assertions.assertNull(eventSearchReq.startTimestamp) { "start timestamp must be null" }
            Assertions.assertNull(eventSearchReq.endTimestamp) { "end timestamp must be null" }
```